p2p/src/net/libp2p/backend.rs

```diff
@@ -101,12 +101,8 @@ impl Libp2pBackend {
                     SwarmEvent::Behaviour(Libp2pBehaviourEvent::Control(
                         ControlEvent::CloseConnection { peer_id })
                     ) => {
-                        // TODO: `inspect_err`
-                        match self.swarm.disconnect_peer_id(peer_id) {
-                            Ok(_) => {}
-                            Err(err) => {
-                                log::error!("Failed to disconnect peer {}: {:?}", peer_id, err);
-                            }
+                        if let Err(e) =  self.swarm.disconnect_peer_id(peer_id) {
+                            log::error!("Failed to disconnect peer {peer_id}: {e:?}");
```

Comment:
why are you not using `tap_err()` here?

Reply:
It is somewhat ironic, because the compiler complains about the unused result. It still can be used this way:
```rust
let _ = self.swarm.disconnect_peer_id(peer_id).tap_err(...);
```
But in that case I prefer to use the `if let Err` construction.

Reply:
Actually I prefer the `let _`, because that signals that the developer deliberately wants to ignore the result, instead of it just being a side-effect.